AppCenterPush/AppCenterPush/MSPush.m

```diff
@@ -118,6 +118,17 @@ + (void *)userNotificationCenterDelegateContext {
 
 #endif
 
+#pragma mark - MSUserIdContextDelegate
+- (void)userIdContext:(MSUserIdContext *)__unused userIdContext didUpdateUserId:(NSString *)userId {
+  NSString *pushTokenCopy = self.pushToken;
+  if (pushTokenCopy) {
+    MSPushLog *log = [MSPushLog new];
+    log.pushToken = pushTokenCopy;
+    log.userId = userId;
+    [self.channelUnit enqueueItem:log flags:MSFlagsDefault];
```

Comment:
1. Reuse send function (add userId in params)
2. Add test for this

Reply:
Good point there should be some code to share here.

Reply:
And it needs testing ;)